src/ram-lib-mapping.ts

```diff
@@ -0,0 +1,520 @@
+  setMinterExtraMinterDetailsValue("numSlots", event.params.numSlots, minter);
+
+  // update minter's updatedAt timestamp to induce a sync, and save
+  minter.updatedAt = event.block.timestamp;
+  minter.save();
+}
+
+export function handleAuctionBufferTimeParamsUpdated(
+  event: AuctionBufferTimeParamsUpdated
+): void {
+  // load minter
+  const minter = loadOrCreateMinter(event.address, event.block.timestamp);
+
+  // update minter extra minter details value
+  setMinterExtraMinterDetailsValue(
+    "auctionBufferSeconds",
+    event.params.auctionBufferSeconds,
+    minter
+  );
+
+  setMinterExtraMinterDetailsValue(
+    "maxAuctionExtraSeconds",
+    event.params.maxAuctionExtraSeconds,
+    minter
+  );
+
+  // update minter's updatedAt timestamp to induce a sync, and save
+  minter.updatedAt = event.block.timestamp;
+  minter.save();
+}
+
+// project-level events
+export function handleAuctionTimestampEndUpdated(
+  event: AuctionTimestampEndUpdated
+): void {
+  const minterProjectAndConfig = loadOrCreateMinterProjectAndConfigIfProject(
+    event.address, // minter
+    event.params.coreContract,
+    event.params.projectId,
+    event.block.timestamp
+  );
+  if (!minterProjectAndConfig) {
+    // project wasn't found, warning already logged in helper function
+    return;
+  }
+
+  const projectMinterConfig = minterProjectAndConfig.projectMinterConfiguration;
+  // update project minter configuration extraMinterDetails json field
+  setProjectMinterConfigExtraMinterDetailsValue(
+    "auctionEndTime",
+    event.params.timestampEnd,
+    projectMinterConfig
+  );
+
+  projectMinterConfig.save();
+
+  // induce sync if the project minter configuration is the active one
+  updateProjectIfMinterConfigIsActive(
+    minterProjectAndConfig.project,
+    projectMinterConfig,
+    event.block.timestamp
+  );
+}
+
+export function handleAuctionConfigUpdated(event: AuctionConfigUpdated): void {
+  const minterProjectAndConfig = loadOrCreateMinterProjectAndConfigIfProject(
+    event.address, // minter
+    event.params.coreContract,
+    event.params.projectId,
+    event.block.timestamp
+  );
+  if (!minterProjectAndConfig) {
+    // project wasn't found, warning already logged in helper function
+    return;
+  }
+
+  const projectMinterConfig = minterProjectAndConfig.projectMinterConfiguration;
+  // update project minter configuration fields
+  projectMinterConfig.priceIsConfigured = true;
+  projectMinterConfig.basePrice = event.params.basePrice;
+  // update all auction details in project minter configuration extraMinterDetails json field
+  let auctionConfigUpdateDetails: TypedMap<
+    string,
+    JSONValue
+  > = createTypedMapFromEntries([
+    {
+      key: "startTime",
+      value: toJSONValue(event.params.timestampStart)
+    },
+    { key: "auctionEndTime", value: toJSONValue(event.params.timestampEnd) },
+    {
+      key: "numTokensInAuction",
+      value: toJSONValue(event.params.numTokensInAuction.toString())
```

Comment:
nit: I think this could just be a number, right? (max of 1 million)

Reply:
Good call! Updated